gc.c

```diff
@@ -8152,18 +8162,10 @@ gc_compact_move(rb_objspace_t *objspace, rb_heap_t *heap, VALUE src)
         return false;
     }
     while (!try_move(objspace, dheap, dheap->free_pages, src)) {
-        struct gc_sweep_context ctx = {
-            .page = dheap->sweeping_page,
-            .final_slots = 0,
-            .freed_slots = 0,
-            .empty_slots = 0,
-        };
-        gc_sweep_page(objspace, heap, &ctx);
-        if (dheap->sweeping_page->free_slots > 0) {
-            heap_add_freepage(dheap, dheap->sweeping_page);
-        };
+        int unlink_pages = GC_SWEEP_PAGES_FREEABLE_PER_STEP;
+        int swept_slots = 0;
+        gc_sweep_page(objspace, pool, dheap->sweeping_page, &unlink_pages, &swept_slots, FALSE);
```

Comment:
This is always going to move the page to the tomb heap if the page is empty because `unlink_pages > 0` is always true.

Reply:
yes, good spot, thanks